pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -249,7 +250,7 @@ pub mod pallet {
 
 	#[pallet::storage]
 	#[pallet::getter(fn pool_with_id)]
-	pub(crate) type Pools<T: Config> = StorageMap<_, Twox64Concat, T::PoolId, PoolDetailsOf<T>, OptionQuery>;
+	pub type Pools<T: Config> = StorageMap<_, Twox64Concat, T::PoolId, PoolDetailsOf<T>, OptionQuery>;
```

Comment:
This should not be made public. You have the getter function that you can use, but let's avoid making these types public.

Reply:
But I also have to iterate over the whole storage. I think this is not possible with the getter

Reply:
Ok, we've done like this in the past and we can do it again. But the best would be to expose a function from the pallet that returns an iterator, so that there cannot be any mistakes where the storage is written to by some other pallet.

Reply:
Yeah, makes sense.

I’ve created a ticket to address this issue: KILTprotocol/ticket#3734.

First, a proper design needs to be developed to handle this across all different pallets, especially when a pallet has multiple maps.

Write
Preview
Cancel
Update comment